web/packages/teleport/src/SessionRecordings/view/Timeline/renderers/TimeMarksRenderer.ts

```diff
@@ -0,0 +1,202 @@
+
+    for (let time = 0; time < this.duration + interval; time += interval) {
+      // if show absolute time is enabled, change the 0:00, 1:00, etc, labels to absolute time
+      const absolute = this.showAbsoluteTime && (time / 1000) % 60 === 0;
+      const label = absolute
+        ? formatAbsoluteTime(this.startTime + time)
+        : formatRelativeTime(time / 1000);
+
+      markers.push({
+        absolute,
+        label,
+        position: (time / this.duration) * this.timelineWidth,
+        time,
+      });
+    }
+
+    const subTicks: SubTick[] = [];
+
+    const markerSpacing = (interval / this.duration) * this.timelineWidth;
+
+    let numSubticks = 0;
+    if (markerSpacing > 300) numSubticks = 9;
+    else if (markerSpacing > 200) numSubticks = 7;
+    else if (markerSpacing > 150) numSubticks = 4;
+    else if (markerSpacing > 100) numSubticks = 3;
+    else if (markerSpacing > 50) numSubticks = 1;
+
+    for (let i = 0; i < markers.length - 1; i++) {
+      const currentMarker = markers[i];
+
+      for (let j = 1; j <= numSubticks; j++) {
+        const fraction = j / (numSubticks + 1);
+        const position = currentMarker.position + markerSpacing * fraction;
+
+        let height = 4;
+        if (numSubticks % 2 === 1) {
+          height = j === Math.ceil(numSubticks / 2) ? 6 : 4;
+        } else {
+          const mid1 = numSubticks / 2;
+          const mid2 = mid1 + 1;
+          height = j === mid1 || j === mid2 ? 6 : 4;
+        }
+
+        subTicks.push({ height, position });
+      }
+    }
+
+    this.timeMarkers = markers;
+    this.subTicks = subTicks;
+  }
+
+  setShowAbsoluteTime(show: boolean) {
+    this.showAbsoluteTime = show;
+
+    this.calculate();
+  }
+
+  private getVisibleTimeMarkers(offset: number, containerWidth: number) {
+    const visibleStart = -offset - 100;
+    const visibleEnd = -offset + containerWidth + 100;
+
+    const visibleMarkers: TimeMarker[] = [];
+    const visibleSubTicks: SubTick[] = [];
+
+    for (const marker of this.timeMarkers) {
+      if (marker.position >= visibleStart && marker.position <= visibleEnd) {
+        visibleMarkers.push(marker);
+      }
+    }
+
+    for (const subTick of this.subTicks) {
+      if (subTick.position >= visibleStart && subTick.position <= visibleEnd) {
+        visibleSubTicks.push(subTick);
+      }
+    }
+
+    return { markers: visibleMarkers, subTicks: visibleSubTicks };
+  }
+}
+
+function formatRelativeTime(seconds: number) {
+  const mins = Math.floor(seconds / 60);
+  const secs = Math.floor(seconds % 60);
+
+  return `${mins}:${secs.toString().padStart(2, '0')}`;
+}
+
+function formatAbsoluteTime(timestamp: number) {
+  const date = new Date(timestamp);
+
+  const hours = date.getHours();
+  const minutes = date.getMinutes();
+
+  const period = hours >= 12 ? 'pm' : 'am';
+  const displayHours = hours % 12 || 12;
+
+  return `${displayHours}:${minutes.toString().padStart(2, '0')}${period}`;
```

Comment:
```suggestion
  return date.toLocaleTimeString([], {hour: '2-digit', minute: '2-digit'})
```

This will honor the browser's locale (not all locales use the 12-hour format).